linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2938,13 +2946,17 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
           })
 
         case JSGlobalRef(name) =>
-          js.VarRef(transformGlobalVarIdent(name))
+          if (name == JSGlobalRef.FileLevelThis)
+            globalVar(VarField.fileLevelThis, CoreVar)
```

Comment:
We define a `$fileLevelThis = this` at the top of the file in `CoreJSLib.defineFileLevelThis`, and `JSGlobalRef(JSGlobalRef.FileLevelThis)` transform to the `$fileLevelThis` (we can think that `$fileLevelThis` as a `globalThis` of ES2020 IIUC).

Why not using `JSGlobalRef("$fileLevelThis")` directly? because we don't wanna allow users to reassign the `$fileLevelThis`. `JSGlobalRef("this")` restricts users from reassigning.

However, with this approach, backends treat `JSGlobalRef("this")` special case which transforms to different.
Maybe we want to add `UnaryOp_fileLevelThis` instead of `JSGlobalRef("this")`?

Reply:
> A UnaryOp needs one operand. Here there would be no operand at all.

Oh, that's right.

> IMO it makes sense for "the global this value" to be represented in a similar way as "the global Math value": in both cases, they represent the value that we would get if we wrote this/Math at the top-level of the generated .js file.

I was thinking that users might want to use `js.Dynamic.global.this` to refer to a receiver object within a method body context, rather than the global object. However, I now think that "representing the value we would get if we wrote `this` or `Math` at the top level of the generated `.js` file" makes sense to me,  since `global.this` resembles `globalThis`  😄